Docs: Create a Full Site Editing overview document

mkaz opened the pull request:
## Description

Creates a new getting started document for full site editng.  There are numerous sections and sub-projects for FSE, this document will gather together these resources and link off so people can find what they need.

@annezazu, I've stubbed out the start to this page and included a user portion to the outreach program.
I would welcome any additional links, sections, comments, details, or anything else you think we can add here.

**Open Question**: I updated "block-based theme" to just "block theme" - from usage I believe this is the direction we want. Do we need to be explict about this? It is easier to not change if we are fine with both.

This addressed concerns and closes #30637


## How has this been tested?

View [rendered document on branch here](https://github.com/WordPress/gutenberg/blob/docs/add-fse-overview/docs/getting-started/full-site-editing.md)

Documentation:

- confirm text makes sense
- check for spelling, typos, phrasing
- confirm links work


## Types of changes

- Adds new Full Site Editing getting-started page
- Renames block-based-theme => block-theme
- Minor tweaks to block-theme tutorial

If we rename, we will want to submit a redirect update to meta trac.

mkaz committed:
Docs: Create a Full Site Editing overview document

Creates a new getting started document for full site editng.
There are numerous sections and sub-projects for FSE, this document will
gather together these resources and link off so people can find what
they need.

Closes #30637

mkaz committed:
Add link to theme.json documentation

carolinan commented:
I am not convinced that the navigation and widget screens are best included here. 
They are a bridge between classic themes and site editing, but they are not available when site editing is fully enabled.

carolinan commented on docs/getting-started/full-site-editing.md:
```diff
@@ -0,0 +1,42 @@
+
+## Get Involved
+
+The majority of full site editing features are avilable in the Gutenberg plungin, and an ongoing [FSE Outreach program](https://make.wordpress.org/test/handbook/full-site-editing-outreach-experiment/) is running with calls for testing amongst a community of folks.
```
```suggestion
The majority of full site editing features are available in the Gutenberg plugin, and an ongoing [FSE Outreach program](https://make.wordpress.org/test/handbook/full-site-editing-outreach-experiment/) is running with calls for testing.
```

carolinan commented on docs/getting-started/full-site-editing.md:
```diff
@@ -0,0 +1,42 @@
+The majority of full site editing features are avilable in the Gutenberg plungin, and an ongoing [FSE Outreach program](https://make.wordpress.org/test/handbook/full-site-editing-outreach-experiment/) is running with calls for testing amongst a community of folks.
+
+- Join in on Slack at #fse-outreach-experiment
+- Participate in the [Calls for Testing](https://make.wordpress.org/test/tag/fse-testing-call/) and test and feedback.
```
```suggestion
- Participate in the [Calls for Testing](https://make.wordpress.org/test/tag/fse-testing-call/), test and give feedback.
```

carolinan commented on docs/getting-started/full-site-editing.md:
```diff
@@ -0,0 +1,42 @@
+
+- Join in on Slack at #fse-outreach-experiment
+- Participate in the [Calls for Testing](https://make.wordpress.org/test/tag/fse-testing-call/) and test and feedback.
+- See the [How to Test FSE instructions](https://make.wordpress.org/test/handbook/full-site-editing-outreach-experiment/how-to-test-fse/) to get setup to test FSE features.
```
I think that setting up FSE needs to be a separate part of the documentation outside of the outreach program.

carolinan replied:
Since it is enough for now, to install a theme with support, and to activate Gutenberg, perhaps this can just be added under the first headline  # Full Site Editing.

mkaz replied:
Updated with TT1 blocks in the directory and moved up instructions to jump in

carolinan commented on docs/getting-started/full-site-editing.md:
```diff
@@ -0,0 +1,42 @@
+
+## Block Themes
+
+If you are using the Gutenberg plugin: you can run, test, and develop block themes. Block themes are themes built using templates composed using blocks.
```
```suggestion
If you are using the Gutenberg plugin you can run, test, and develop block themes. Block themes are themes built using templates composed using blocks.
```

carolinan replied:
_Block themes are themes built using templates composed using blocks._

I think we can do better here, this sentence reads a bit strange to me.

carolinan commented on docs/getting-started/full-site-editing.md:
```diff
@@ -0,0 +1,42 @@
+
+- See the [Create a Block Theme](/docs/how-to-guides/block-theme/README.md) tutorial for a walk-through of the pieces of a block theme.
+
+- If you want to dive right in, see the [github.com/WordPress/theme-experiments](https://github.com/WordPress/theme-experiments/) repository. You can download different example block themes there, try the [TT1 Blocks](https://github.com/WordPress/theme-experiments/tree/master/tt1-blocks) is a block theme recreating the TwentyTwentyOne default theme.
```
I think it is important to mention that TT1 Blocks can be downloaded from the theme directory.

mkaz replied:
Agreed, great suggestion! I'll add it.

carolinan replied:
Actually, all current FSE themes that are in the theme directory can be found with the feature filter:
https://wordpress.org/themes/tags/full-site-editing/

mkaz replied:
Added :+1:

mkaz committed:
Apply suggestions from code review

Thank you for fixing my embarrassing typos :-)

Co-authored-by: Carolina Nymark <myazalea@hotmail.com>

carolinan commented on docs/getting-started/full-site-editing.md:
```diff
@@ -0,0 +1,42 @@
+
+- If you want to dive right in, see the [github.com/WordPress/theme-experiments](https://github.com/WordPress/theme-experiments/) repository. You can download different example block themes there, try the [TT1 Blocks](https://github.com/WordPress/theme-experiments/tree/master/tt1-blocks) is a block theme recreating the TwentyTwentyOne default theme.
+
+- Use the `empty-theme.php` script from theme-experiments repo to generate a start block theme, it will prompt you with a few questions and create a theme.
```
🤔  This has not been updated since December, **as far as I remember** it does not include the block-templates and block-template-parts directories or theme.json. It needs to be tested to assure that it is up to date before this is included in the docs.

carolinan commented on docs/how-to-guides/block-theme/README.md:
```diff
@@ -1,36 +1,32 @@
 ### Required files and file structure
 
-A block based theme requires an `index.php` file, an index template file, a `style.css` file, and a `functions.php` file.
+A block theme requires an `index.php` file, an index template file, a `style.css` file, and a `functions.php` file.
```
This is not 100% true but lets update this when the entire tutorial can be updated, after the changes to theme.json.

mkaz replied:
Yes, agree I think we can do another pass on the tutorial. :+1:  

I was just updating to move towards the "block theme" usage instead of "block based theme"

carolinan commented on docs/getting-started/full-site-editing.md:
```diff
@@ -0,0 +1,42 @@
+
+- See the [Create a Block Theme](/docs/how-to-guides/block-theme/README.md) tutorial for a walk-through of the pieces of a block theme.
+
+- If you want to dive right in, see the [github.com/WordPress/theme-experiments](https://github.com/WordPress/theme-experiments/) repository. You can download different example block themes there, try the [TT1 Blocks](https://github.com/WordPress/theme-experiments/tree/master/tt1-blocks) is a block theme recreating the TwentyTwentyOne default theme.
```
```suggestion
- If you want to dive right in, see the [github.com/WordPress/theme-experiments](https://github.com/WordPress/theme-experiments/) repository. You can download different example block themes there, try the [TT1 Blocks](https://github.com/WordPress/theme-experiments/tree/master/tt1-blocks) is a block theme recreating the Twenty Twenty-One default theme.
```

carolinan committed:
Update docs/getting-started/full-site-editing.md

carolinan commented:
It might be out of scope but I think this page should be removed https://developer.wordpress.org/block-editor/how-to-guides/block-based-theme/block-based-themes-2-adding-blocks/

carolinan commented:
There is also this page about templates that could be moved to an easier to find place:
https://developer.wordpress.org/block-editor/explanations/architecture/full-site-editing-templates/

mkaz commented:
@carolinan I also noticed that one of the block theme pages isn't showing up in the directory. I'll try to get this cleaned up but I might do in a couple of different PRs to focus the work.

mkaz committed:
Update with TT1 Block theme available in directory

mkaz committed:
Reword jump in section, add link to tag

mkaz committed:
Add link to templates architecture doc

mkaz committed:
Remove outdated adding blocks to block theme page

mkaz committed:
Add link to block theme overview

mkaz commented:
I removed the adding blocks page, which I think will fix the currently unlinkable block theme overview, which I also added a link to in the overview document.

The tutorial can use updating, please the block theme overview, but I think we can leave those for other PRs. So this should be ready to ship 🚢

annezazu commented:
> Open Question: I updated "block-based theme" to just "block theme" - from usage I believe this is the direction we want. Do we need to be explicit about this? It is easier to not change if we are fine with both.

Yes, I think we should! Great snag. Reviewing the rest of the doc now.

annezazu committed:
Edits to FSE documentation

- Cleaned up some of the phrasing
- Added in a few more links
- Made it clear to test with a test site 
- Added context around nav/widgets + the vision for FSE

carolinan reviewed (approved):
I am not sure how to test the link changes, but I think the content is ready.


But I wonder if there really needs to be both a theme and block theme section under how-to guides, or if this could be combined under _Themes_.
I am not that found of the "Theming for the Block Editor" heading.
